examples/mongocxx/mongodb.com/change_streams_examples.cpp

```diff
@@ -0,0 +1,218 @@
+    }
+}
+
+void change_streams_example_3(mongocxx::collection inventory) {
+    bsoncxx::stdx::optional<bsoncxx::document::value> next;
+
+    // Get one notification to set `next`.
+    {
+        change_stream stream = inventory.watch();
+        auto it = stream.begin();
+        while (it == stream.end()) {
+            // Server returned no new notifications. Restart iteration to poll server.
+            it = stream.begin();
+        }
+        next = bsoncxx::document::value(*it);
+    }
+
+    // Start Changestream Example 3
+    auto resume_token = (*next)["_id"].get_document().value;
+    options::change_stream options;
+    options.resume_after(resume_token);
+    change_stream stream = inventory.watch(options);
+    auto it = stream.begin();
+    while (it == stream.end()) {
+        // Server returned no new notifications. Restart iteration to poll server.
+        it = stream.begin();
+    }
+    // End Changestream Example 3
+
+    if ((*it)["operationType"].get_string().value != "insert") {
+        throw std::logic_error{"expected operationType to be 'insert'"};
+    }
+}
+
+void change_streams_example_4(mongocxx::collection inventory) {
+    // Create a pipeline with
+    //  [{"$match": {"$or": [{"fullDocument.username": "alice"}, {"operationType": "delete"}]}}]
+
+    // Start Changestream Example 4
+    mongocxx::pipeline cs_pipeline;
+    cs_pipeline.match(make_document(
+        kvp("$or",
+            make_array(
+                make_document(kvp("fullDocument.username", "alice")), make_document(kvp("operationType", "delete"))))));
+
+    change_stream stream = inventory.watch(cs_pipeline);
+    auto it = stream.begin();
+    while (it == stream.end()) {
+        // Server returned no new notifications. Restart iteration to poll server.
+        it = stream.begin();
+    }
+    // End Changestream Example 4
+
+    if ((*it)["operationType"].get_string().value != "insert") {
+        throw std::logic_error{"expected operationType to be 'insert'"};
+    }
+}
+
+} // namespace
+
+int EXAMPLES_CDECL main() {
+    if (char const* const topology_env = std::getenv("MONGOCXX_TEST_TOPOLOGY")) {
+        auto const topology = std::string(topology_env);
+        if (topology != "replica") {
+            std::cerr << "Skipping: change_streams example requires a replica set" << std::endl;
+            return 0;
+        }
+    }
+
+    // The mongocxx::instance constructor and destructor initialize and shut down the driver,
+    // respectively. Therefore, a mongocxx::instance must be created before using the driver and
+    // must remain alive for as long as the driver is in use.
+    mongocxx::instance const inst{};
+
+    std::function<void(mongocxx::collection)> const examples[] = {
+        change_streams_example_1,
+        change_streams_example_2,
+        change_streams_example_3,
+        change_streams_example_4,
+    };
+
+    for (auto const& example : examples) {
+        mongocxx::pool pool{mongocxx::uri{}};
```

Comment:
```suggestion
    mongocxx::pool pool{mongocxx::uri{}};
    for (auto const& example : examples) {
```

Suggest creating pool outside of loop. Pool destruction signals and waits for monitoring threads to shut down. Timing `cmake --build cmake-build --target run-examples-mongocxx-mongodb.com-change_streams_examples` locally showed a decrease of 2.207s => 1.156s.